### ISSUE TITLE
runtime panic on file-server

### ISSUE THREAD

**Issue by vired** [caddyserver/caddy#3259]
caddy version - `v2.0.0-rc.2 h1:7NOaxYrsnQ5D3rcLGn2UnLZHLfBpgrfM/JNTLhjCJ1c=`

commad - `caddy file-server --listen localhost:8000 --browse`
 
at startup
```
2020/04/13 07:33:41.118	WARN	admin	admin endpoint disabled
2020/04/13 13:33:41 [INFO][cache:0xc0005a8820] Started certificate maintenance routine
2020/04/13 07:33:41.121	INFO	tls	cleaned up storage units
2020/04/13 07:33:41.264	INFO	autosaved config	{"file": "/home/ahmed/.config/caddy/autosave.json"}
2020/04/13 13:33:41 Caddy 2 serving static files on localhost:8000
```

when i visit localhost:8000 via firefox, the page loads but server panic
```
2020/04/13 13:37:32 http: panic serving 127.0.0.1:36188: runtime error: invalid memory address or nil pointer dereference
goroutine 22 [running]:
net/http.(*conn).serve.func1(0xc000149040)
	net/http/server.go:1772 +0x139
panic(0x144d380, 0x2470800)
	runtime/panic.go:975 +0x3e3
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc0001a6c60, 0x192e2e0, 0xc00066e1c0, 0xc0001e1b00)
	github.com/caddyserver/caddy/v2@v2.0.0-rc.2/modules/caddyhttp/server.go:203 +0x932
net/http.serverHandler.ServeHTTP(0xc00066e000, 0x192e2e0, 0xc00066e1c0, 0xc0000b4200)
	net/http/server.go:2807 +0xa3
net/http.(*conn).serve(0xc000149040, 0x19334e0, 0xc0004941c0)
	net/http/server.go:1895 +0x86c
created by net/http.(*Server).Serve
	net/http/server.go:2933 +0x35c
2020/04/13 13:37:32 http: panic serving 127.0.0.1:36190: runtime error: invalid memory address or nil pointer dereference
goroutine 14 [running]:
net/http.(*conn).serve.func1(0xc0004d0fa0)
	net/http/server.go:1772 +0x139
panic(0x144d380, 0x2470800)
	runtime/panic.go:975 +0x3e3
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc0001a6c60, 0x192e2e0, 0xc000d72000, 0xc00069e700)
	github.com/caddyserver/caddy/v2@v2.0.0-rc.2/modules/caddyhttp/server.go:203 +0x932
net/http.serverHandler.ServeHTTP(0xc00066e000, 0x192e2e0, 0xc000d72000, 0xc00069e500)
	net/http/server.go:2807 +0xa3
net/http.(*conn).serve(0xc0004d0fa0, 0x19334e0, 0xc000653ec0)
	net/http/server.go:1895 +0x86c
created by net/http.(*Server).Serve
	net/http/server.go:2933 +0x35c
2020/04/13 13:37:32 http: panic serving 127.0.0.1:36192: runtime error: invalid memory address or nil pointer dereference
goroutine 51 [running]:
net/http.(*conn).serve.func1(0xc0001490e0)
	net/http/server.go:1772 +0x139
panic(0x144d380, 0x2470800)
	runtime/panic.go:975 +0x3e3
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc0001a6c60, 0x192e2e0, 0xc00066e2a0, 0xc001766100)
	github.com/caddyserver/caddy/v2@v2.0.0-rc.2/modules/caddyhttp/server.go:203 +0x932
net/http.serverHandler.ServeHTTP(0xc00066e000, 0x192e2e0, 0xc00066e2a0, 0xc0001e1c00)
	net/http/server.go:2807 +0xa3
net/http.(*conn).serve(0xc0001490e0, 0x19334e0, 0xc00067e1c0)
	net/http/server.go:1895 +0x86c
created by net/http.(*Server).Serve
	net/http/server.go:2933 +0x35c
2020/04/13 13:37:32 http: panic serving 127.0.0.1:36194: runtime error: invalid memory address or nil pointer dereference
goroutine 16 [running]:
net/http.(*conn).serve.func1(0xc0004d1900)
	net/http/server.go:1772 +0x139
panic(0x144d380, 0x2470800)
	runtime/panic.go:975 +0x3e3
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc0001a6c60, 0x192e2e0, 0xc000d720e0, 0xc00069ea00)
	github.com/caddyserver/caddy/v2@v2.0.0-rc.2/modules/caddyhttp/server.go:203 +0x932
net/http.serverHandler.ServeHTTP(0xc00066e000, 0x192e2e0, 0xc000d720e0, 0xc00069e800)
	net/http/server.go:2807 +0xa3
net/http.(*conn).serve(0xc0004d1900, 0x19334e0, 0xc0006fe140)
	net/http/server.go:1895 +0x86c
created by net/http.(*Server).Serve
	net/http/server.go:2933 +0x35c
2020/04/13 13:37:32 http: panic serving 127.0.0.1:36196: runtime error: invalid memory address or nil pointer dereference
goroutine 53 [running]:
net/http.(*conn).serve.func1(0xc000149180)
	net/http/server.go:1772 +0x139
panic(0x144d380, 0x2470800)
	runtime/panic.go:975 +0x3e3
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc0001a6c60, 0x192e2e0, 0xc000d721c0, 0xc00069ec00)
	github.com/caddyserver/caddy/v2@v2.0.0-rc.2/modules/caddyhttp/server.go:203 +0x932
net/http.serverHandler.ServeHTTP(0xc00066e000, 0x192e2e0, 0xc000d721c0, 0xc001766200)
	net/http/server.go:2807 +0xa3
net/http.(*conn).serve(0xc000149180, 0x19334e0, 0xc00067e3c0)
	net/http/server.go:1895 +0x86c
created by net/http.(*Server).Serve
	net/http/server.go:2933 +0x35c
2020/04/13 13:37:32 http: panic serving 127.0.0.1:36198: runtime error: invalid memory address or nil pointer dereference
goroutine 54 [running]:
net/http.(*conn).serve.func1(0xc000149220)
	net/http/server.go:1772 +0x139
panic(0x144d380, 0x2470800)
	runtime/panic.go:975 +0x3e3
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc0001a6c60, 0x192e2e0, 0xc000d722a0, 0xc00069ef00)
	github.com/caddyserver/caddy/v2@v2.0.0-rc.2/modules/caddyhttp/server.go:203 +0x932
net/http.serverHandler.ServeHTTP(0xc00066e000, 0x192e2e0, 0xc000d722a0, 0xc00069ed00)
	net/http/server.go:2807 +0xa3
net/http.(*conn).serve(0xc000149220, 0x19334e0, 0xc0006fe3c0)
	net/http/server.go:1895 +0x86c
created by net/http.(*Server).Serve
	net/http/server.go:2933 +0x35c
2020/04/13 13:37:32 http: panic serving 127.0.0.1:36200: runtime error: invalid memory address or nil pointer dereference
goroutine 55 [running]:
net/http.(*conn).serve.func1(0xc0001492c0)
	net/http/server.go:1772 +0x139
panic(0x144d380, 0x2470800)
	runtime/panic.go:975 +0x3e3
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc0001a6c60, 0x192e2e0, 0xc00066e380, 0xc001766500)
	github.com/caddyserver/caddy/v2@v2.0.0-rc.2/modules/caddyhttp/server.go:203 +0x932
net/http.serverHandler.ServeHTTP(0xc00066e000, 0x192e2e0, 0xc00066e380, 0xc001766300)
	net/http/server.go:2807 +0xa3
net/http.(*conn).serve(0xc0001492c0, 0x19334e0, 0xc00067e440)
	net/http/server.go:1895 +0x86c
created by net/http.(*Server).Serve
	net/http/server.go:2933 +0x35c
2020/04/13 13:37:32 http: panic serving 127.0.0.1:36202: runtime error: invalid memory address or nil pointer dereference
goroutine 23 [running]:
net/http.(*conn).serve.func1(0xc0001a06e0)
	net/http/server.go:1772 +0x139
panic(0x144d380, 0x2470800)
	runtime/panic.go:975 +0x3e3
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc0001a6c60, 0x192e2e0, 0xc000c1c000, 0xc000c18200)
	github.com/caddyserver/caddy/v2@v2.0.0-rc.2/modules/caddyhttp/server.go:203 +0x932
net/http.serverHandler.ServeHTTP(0xc00066e000, 0x192e2e0, 0xc000c1c000, 0xc000c18000)
	net/http/server.go:2807 +0xa3
net/http.(*conn).serve(0xc0001a06e0, 0x19334e0, 0xc000122340)
	net/http/server.go:1895 +0x86c
created by net/http.(*Server).Serve
	net/http/server.go:2933 +0x35c
2020/04/13 13:37:32 http: panic serving 127.0.0.1:36204: runtime error: invalid memory address or nil pointer dereference
goroutine 57 [running]:
net/http.(*conn).serve.func1(0xc000149360)
	net/http/server.go:1772 +0x139
panic(0x144d380, 0x2470800)
	runtime/panic.go:975 +0x3e3
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc0001a6c60, 0x192e2e0, 0xc00066e460, 0xc001766800)
	github.com/caddyserver/caddy/v2@v2.0.0-rc.2/modules/caddyhttp/server.go:203 +0x932
net/http.serverHandler.ServeHTTP(0xc00066e000, 0x192e2e0, 0xc00066e460, 0xc001766600)
	net/http/server.go:2807 +0xa3
net/http.(*conn).serve(0xc000149360, 0x19334e0, 0xc00067e5c0)
	net/http/server.go:1895 +0x86c
created by net/http.(*Server).Serve
	net/http/server.go:2933 +0x35c
2020/04/13 13:37:32 http: panic serving 127.0.0.1:36206: runtime error: invalid memory address or nil pointer dereference
goroutine 25 [running]:
net/http.(*conn).serve.func1(0xc0001a0780)
	net/http/server.go:1772 +0x139
panic(0x144d380, 0x2470800)
	runtime/panic.go:975 +0x3e3
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc0001a6c60, 0x192e2e0, 0xc000c1c0e0, 0xc000c18500)
	github.com/caddyserver/caddy/v2@v2.0.0-rc.2/modules/caddyhttp/server.go:203 +0x932
net/http.serverHandler.ServeHTTP(0xc00066e000, 0x192e2e0, 0xc000c1c0e0, 0xc000c18300)
	net/http/server.go:2807 +0xa3
net/http.(*conn).serve(0xc0001a0780, 0x19334e0, 0xc0003d2d80)
	net/http/server.go:1895 +0x86c
created by net/http.(*Server).Serve
	net/http/server.go:2933 +0x35c

```

but if i use `curl localhost:8000` there is no panic.

i also tested on brave, it panic.

```
uname -a
Linux localhost.localdomain 5.6.0-0.rc5.git0.2.fc32.x86_64 #1 SMP Tue Mar 10 19:09:42 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

```

**Comment by mlaily** [comment]
Same problem here.
Loading an image for example, triggers a panic (but the image loads successfully in the browser).
Example: https://docs.matrix.x2a.yt/screenshots/1a-login.png
```
2020/04/13 08:05:49 http2: panic serving 109.26.2.233:16090: runtime error: invalid memory address or nil pointer dereference
goroutine 8255 [running]:
net/http.(*http2serverConn).runHandler.func1(0xc00000e438, 0xc000783f8e, 0xc00019ef00)
        net/http/h2_bundle.go:5713 +0x16b
panic(0x144d380, 0x2470800)
        runtime/panic.go:969 +0x166
github.com/caddyserver/caddy/v2/modules/caddyhttp.(*Server).ServeHTTP(0xc0003c2120, 0x192e1e0, 0xc00000e438, 0xc0022f1f00)
        github.com/caddyserver/caddy/v2@v2.0.0-rc.2/modules/caddyhttp/server.go:203 +0x932
net/http.serverHandler.ServeHTTP(0xc0005fae00, 0x192e1e0, 0xc00000e438, 0xc000891700)
        net/http/server.go:2807 +0xa3
net/http.initALPNRequest.ServeHTTP(0x19335a0, 0xc0007cc540, 0xc0004e7880, 0xc0005fae00, 0x192e1e0, 0xc00000e438, 0xc000891700)
        net/http/server.go:3381 +0x8d
net/http.(*http2serverConn).runHandler(0xc00019ef00, 0xc00000e438, 0xc000891700, 0xc0006ea8c0)
        net/http/h2_bundle.go:5720 +0x8b
created by net/http.(*http2serverConn).processHeaders
        net/http/h2_bundle.go:5454 +0x4e1
```

The complete configuration for this hostname looks like this (Some parts are imported from another file):
```
docs.matrix.x2a.yt {
        encode gzip

        rewrite / /index.htm

        root * /srv/www/docs.matrix.x2a.yt/
        file_server
}
```

**Comment by Henrocker** [comment]
@vired @mlaily This is already fixed in 999ab22 I believe.

Feel free to download an updated version of Caddy v2 here: [Linux](https://github.com/caddyserver/caddy/suites/590488726/artifacts/4224887), [Windows](https://github.com/caddyserver/caddy/suites/590488726/artifacts/4224889) and [macOS](https://github.com/caddyserver/caddy/suites/590488726/artifacts/4224888).

**Comment by francislavoie** [comment]
As @Henrocker said, this is a duplicate issue, already fixed. Please use the latest CI artifact for the time being, until the next release. Thanks!
